### PR TITLE
Add go 1.19, require go 1.18, drop go 1.17

### PR DESCRIPTION
**Pull request by kolyshkin**
This makes go 1.18 a minimally required version, removing go 1.17 from and adding go 1.19 to CI.

Also:
 - [x] remove old-style `+build` tags
 - [x] fix the docstrings to be compatible with go 1.19 (see https://tip.golang.org/doc/comment)
 - [x] wait for golangci-lint release that fixes https://github.com/golangci/golangci-lint/issues/2922
 - [x] remove `test (1.17.x)` from the required set of tests
 - [ ] add `test (1.19.x)` to required list of tests (once this is merged)

### CODE REVIEW

**Comment by kolyshkin** [comment]
OK, it seems to be working fine, except for golangci-lint (filed https://github.com/golangci/golangci-lint/issues/2922)

**Comment by AkihiroSuda** [comment]
1.19 is GA now

**Comment by kolyshkin** [comment]
Alas we have to wait for a new golangci-lint release that has https://github.com/golangci/golangci-lint/issues/2922 fixed. The fix is already merged but no release has been made yet (I guess it will be v1.48).

**Comment by kolyshkin** [comment]
Since I've already marked 1.19.x CI tests as required, can we please merge this? @opencontainers/runc-maintainers PTAL

File: libcontainer/apparmor/apparmor_unsupported.go
Comment (thaJeztah) [inline libcontainer/apparmor/apparmor_unsupported.go:2]: I'm on the fence on already dropping these, as we know libcontainer is still used by external projects, and those may not yet have updated to go 1.18+ (I saw there was some discussion among containerd maintainers as well)

I guess it slightly depends on which release this will get into, and overall it seems ok to (start) dropping 1.17 in CI, but just on the fence on doing a hard _break_ for Go < 1.18

Comment (kolyshkin) [reply libcontainer/apparmor/apparmor_unsupported.go:2]: The tags removal is done deliberately so the codebase won't work with older golang. This way we can start using new(ish) features of golang 1.18.

I was more conservative about it in the past, supporting three golang releases (including the latest unsupported version), but now when we have branches, we can always say "if you want older golang, use older runc/libcontainer). To me it seems a nice balance between using new golang features and supporting older releases.

So, this code is in main, meaning it's going to be 1.2.0, which we will hopefully release in September/October. From what I see, k8s (I guess this is the biggest user of runc) is switching to newest golang very aggressively.

Comment (kolyshkin) [reply libcontainer/apparmor/apparmor_unsupported.go:2]: > overall it seems ok to (start) dropping 1.17 in CI, but just on the fence on doing a hard break for Go < 1.18

Doesn't work this way. Once we drop Go 1.17 and someone will add e.g. use of `strings.Cut`, we'll never know that the code is broken for go < 1.18 until someone will use it (after the release). So it's either we support it (and test it in CI), or we don't (and we make this deliberate). Having something in the middle doesn't seem like a good way to me.

Comment (kolyshkin) [reply libcontainer/apparmor/apparmor_unsupported.go:2]: @thaJeztah PTAL ^^

Comment (kolyshkin) [reply libcontainer/apparmor/apparmor_unsupported.go:2]: @AkihiroSuda what do you think?

Comment (thaJeztah) [reply libcontainer/apparmor/apparmor_unsupported.go:2]: My main concern is that `<unknown>` project or distro may still be using older Go versions (I recall discussions about Debian etc not updating).

And while we no longer test on those versions, making things a fully breaking change (not even possible to compile) a bit harsh, and _possibly_ even would require a major version update (from a SemVer perspective); I know Golang's own packages, such as `golang.org/x/sys` continue to include both old and new-style build-tags; https://github.com/golang/sys/blob/1c4a2a72c664c42691e45fe3a31510ec5ba13cfb/unix/str.go#L5-L6

Comment (kolyshkin) [reply libcontainer/apparmor/apparmor_unsupported.go]: Removed the last commit; PTAL @thaJeztah @AkihiroSuda

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by kolyshkin** [comment]
@AkihiroSuda PTAL 🙏🏻

**Review by AkihiroSuda — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
@kolyshkin are you updating the repo settings? ("add test (1.19.x) to required list of tests (once this is merged)")

**Pull request merged**
